docs/designers-developers/developers/tutorials/format-api/3-apply-format.md

```diff
@@ -0,0 +1,43 @@
+# Apply the format when the button is clicked
+
+So far, your custom button doesn't modify the text selected, it only renders a message in the console. Let's change that.
+
+The [rich-text package](/packages/rich-text/README.md) offers a few utilities to work with formats: [applyFormat](/packages/rich-text/README.md#applyFormat), [removeFormat](/packages/rich-text/README.md#removeFormat), and [toggleFormat](/packages/rich-text/README.md#toggleFormat). In this particular example, the format you want to apply (the `<samp>` tag) may be considered binary -either a text selection has the tag or hasn't. Taking that into account, the `toggleFormat` primitive seems smore convenient.
+
+Update `my-custom-format.js` with this new code:
+
+```js
+( function( wp ) {
+	var MyCustomButton = function( props ) {
+		return wp.element.createElement(
+			wp.editor.RichTextToolbarButton, {
+				icon: 'editor-code',
+				title: 'Sample output',
+				onClick: function() {
+					props.onChange( wp.richText.toggleFormat(
+						props.value,
+						{ type: 'my-custom-format/sample-output' }
+					) );
+				},
+				isActive: props.isActive,
+			}
+		);
+	}
+	wp.richText.registerFormatType(
+		'my-custom-format/sample-output', {
+			title: 'Sample output',
+			tagName: 'samp',
+			className: null,
+			edit: MyCustomButton,
+		}
+	);
+} )( window.wp );
+```
+
+Now, let's check that is working as intended: reload the post/page, make a text selection, click the button, and then change to HTML view to confirm that the tag was effectively applied.
+
+The expected behavior is that the format will be toggled, meaning that the text selected will be wrapped by a `<samp>` tag if it isn't yet, or the tag will be removed if the selection is already wrapped with the tag. Notice that the button renders a different style depending on whether the selection has the tag or not as well - this is controlled by the `isActive` property of the `RichTextToolbarButton` component.
+
+Probably your browser has already displayed the selection differently once the tag was applied. You may want to use a special style for it, though. So far, you can only target the `samp` element in your CSS to do so. Another option that you could explore is to add a class to the format using the `className` option in [`registerFormatType`](/packages/rich-text/README.md#registerFormatType).
```

Comment:
```suggestion
Your browser may have already displayed the selection differently once the tag was applied, but you may want to use a special style of your own. So far, you can only target the `samp` element in your CSS to do so. Another option that you could explore is to add a class to the format using the `className` option in [`registerFormatType`](/packages/rich-text/README.md#registerFormatType).
```

Reply:
It's also super simple but a CSS code block might be nice here. I also wonder if it's even worth building the `className` support into the tutorial, since it'd be good practice to encourage className scoping?

Reply:
How about updating the text to communicate that clearer?

```suggestion
Your browser may have already displayed the selection differently once the tag was applied, but you may want to use a special style of your own. If the `className` property in [`registerFormatType`](/packages/rich-text/README.md#registerFormatType) is not `null`, the declared class will be added to the new element, so you can target it by class name.
```

Reply:
Commited a new version.